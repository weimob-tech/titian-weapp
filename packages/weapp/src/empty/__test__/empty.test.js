const simulate = require('miniprogram-simulate');
const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', async () => {
    const comp = getComponent('empty');
    const clickCallback = jest.fn();
    comp.addEventListener('click', clickCallback);

    expect(comp).toMatchSnapshot();

    comp.querySelector('.titian-empty').dispatchEvent('tap');
    await simulate.sleep(0);

    expect(clickCallback).toHaveBeenCalledTimes(1);
  });
});
