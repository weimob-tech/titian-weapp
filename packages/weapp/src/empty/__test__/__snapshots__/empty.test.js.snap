// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="titian-empty titian-empty-medium ext-class"
    style=""
    bind:tap="onClick"
  >
    <wx-image
      class="titian-empty-image"
      src="https://cdn2.weimob.com/saas/saas-fe-sirius-orion-node/production/145/searchwithnoresult.png"
    />
  </wx-view>
</main>
`;
